### ISSUE TITLE
sarltrain

### ISSUE THREAD

**Issue by johnwillan** [huangkk99/Meta-RL-for-fast-spectrum-sharing#5]
老师您好，在sample方法里a_prob = torch.cat(l_r, dim=0).unsqueeze(1).to(device)
使用了 l_r （奖励）而不是 l_a_p （动作概率），这是什么原因？
还有就是Environment_marl.py 中，如果 lambdda = 0. ，那么训练时完全忽略了V2I通信速率？
我将此代码扩展为MAPPO时，测试性能很差

![Image](https://github.com/user-attachments/assets/4da00146-8fe8-4143-9dc3-7c10de35b633)

**Comment by huangkk99** [comment]
不好意思，这里是个错误，确实应该是a_prob = torch.cat(l_a_p, dim=0).unsqueeze(1).to(device)；
不过可能由于该环境的问题，误写为l_r的性能也比较好。在你的环境中最好改成正确的形式。
lambda=0.的设定参考https://github.com/le-liang/MARLspectrumSharingV2X ，通过实验发现这里设为0时获得的效果最好，因此延用了这样的设定。

**Comment by johnwillan** [comment]
老师您好，我想请教一下是不是在这个车联网环境下PPO算法性能似乎不如DQN算法？我试着将梁老师和您的这两篇论文改写为MAPPO版本，虽然reward是收敛的，但是V2Irate和V2V success甚至不如基线水平。
